### PR TITLE
[SDK-547] upgrade gradle & JacksonJaxbJsonProvider

### PR DESCRIPTION
**Pull request by shy950521**

### CODE REVIEW

File: gradle/wrapper/gradle-wrapper.properties [hunk 1]
```diff
@@ -1,6 +1,6 @@
-#Thu Mar 02 11:15:41 CST 2017
+#Wed Mar 24 10:47:53 CST 2021
 distributionBase=GRADLE_USER_HOME
 distributionPath=wrapper/dists
 zipStoreBase=GRADLE_USER_HOME
 zipStorePath=wrapper/dists
-distributionUrl=https\://services.gradle.org/distributions/gradle-2.14-bin.zip
+distributionUrl=https\://services.gradle.org/distributions/gradle-5.6.4-bin.zip
```
Comment (twincitiesguy) [inline gradle/wrapper/gradle-wrapper.properties:6]: This is a big enough change (v2 to v5) that justifies updating all of the gradle wrapper components.  Do this by running `./gradlew wrapper --gradle-version 5.6.4`.  Run this **twice** to make sure the jar gets updated too - then commit all of the wrapper artifacts (including the jar - might have to use `git add --force`).

File: build.gradle [hunk 2]
```diff
@@ -24,29 +24,242 @@
...
-    ext.commonBuildDir = "https://raw.githubusercontent.com/emcvipr/ecs-common-build/v$commonBuildVersion"
-    apply from: "$commonBuildDir/ecs-publish.buildscript.gradle", to: buildscript
-}
+defaultTasks 'distZip'
 
-apply from: "$commonBuildDir/ecs-publish.gradle"
+repositories {
+    mavenCentral()
+    mavenLocal()
+    maven {
```
Comment (twincitiesguy) [inline build.gradle:59]: why was this repo added?  everything should be in maven central

File: build.gradle [hunk 3]
```diff
@@ -24,29 +24,242 @@
...
+        url "https://maven.atlassian.com/3rdparty/"
+    }
+}
 
 dependencies {
-    compile 'com.sun.jersey:jersey-client:1.19.4',
-            'com.sun.jersey.contribs:jersey-apache-client4:1.19.4',
-            'org.codehaus.jackson:jackson-jaxrs:1.9.13',
-            'org.codehaus.jackson:jackson-xc:1.9.13',
+    implementation 'log4j:log4j:1.2.17',
```
Comment (twincitiesguy) [inline build.gradle:65]: remove log4j from compile dependencies - we use slf4j so users can choose their own logger

File: build.gradle [hunk 4]
```diff
@@ -24,29 +24,242 @@
...
+    implementation ('com.sun.jersey:jersey-json:1.19.4') {
         exclude group: 'org.codehaus.jackson'
     }
-    runtime 'org.slf4j:slf4j-log4j12:1.7.5'
-    testCompile 'junit:junit:4.12',
-            'log4j:log4j:1.2.17'
+    implementation 'com.sun.jersey.contribs:jersey-apache-client4:1.19.4'
+
+    runtimeOnly 'org.slf4j:slf4j-log4j12:1.7.5'
+    testImplementation 'junit:junit:4.12'
```
Comment (twincitiesguy) [inline build.gradle:75]: tests need log4j, that's why it was testCompile previously

File: build.gradle [hunk 5]
```diff
@@ -24,29 +24,242 @@
...
-    compile 'com.sun.jersey:jersey-client:1.19.4',
-            'com.sun.jersey.contribs:jersey-apache-client4:1.19.4',
-            'org.codehaus.jackson:jackson-jaxrs:1.9.13',
-            'org.codehaus.jackson:jackson-xc:1.9.13',
+    implementation 'log4j:log4j:1.2.17',
             'org.apache.httpcomponents:httpclient:4.2.6',
-            'org.slf4j:slf4j-api:1.7.5'
-    compile('com.sun.jersey:jersey-json:1.19.4') {
+            'org.slf4j:slf4j-api:1.7.5',
+            'com.fasterxml.jackson.jaxrs:jackson-jaxrs-json-provider:2.12.2'
```
Comment (twincitiesguy) [inline build.gradle:68]: I _think_ we also need `com.fasterxml.jackson.module:jackson-module-jaxb-annotations:2.12.2`

**Review by twincitiesguy — CHANGES_REQUESTED** [review]

File: build.gradle [hunk 6]
```diff
@@ -24,29 +24,242 @@
...
+    id 'nebula.release' version '15.2.0'
+}
+
+group 'com.emc.ecs'
 description = 'Smart REST Client - JAX-RS (Jersey) REST client that provides client-side load balancing with a pluggable host list provider'
 
+// name of the github project repository
 ext.githubProjectName = 'smart-client-java'
+// URL to github project
+ext.githubProjectUrl = "https://github.com/twincitiesguy/${githubProjectName}"
```
[the remark this answers is not in the document]
Comment (shy950521) [reply build.gradle]: done

File: gradle/wrapper/gradle-wrapper.properties (hunk 1 above)
Comment (shy950521) [reply gradle/wrapper/gradle-wrapper.properties]: done

File: build.gradle (hunk 2 above)
Comment (shy950521) [reply build.gradle]: done

File: build.gradle (hunk 3 above)
Comment (shy950521) [reply build.gradle]: done

File: build.gradle (hunk 4 above)
Comment (shy950521) [reply build.gradle]: done

File: build.gradle (hunk 5 above)
Comment (shy950521) [reply build.gradle]: done

File: build.gradle [hunk 7]
```diff
@@ -55,24 +55,21 @@ defaultTasks 'distZip'
...
 }
 
 dependencies {
-    implementation 'log4j:log4j:1.2.17',
-            'org.apache.httpcomponents:httpclient:4.2.6',
+    implementation 'org.apache.httpcomponents:httpclient:4.2.6',
             'org.slf4j:slf4j-api:1.7.5',
-            'com.fasterxml.jackson.jaxrs:jackson-jaxrs-json-provider:2.12.2'
+            'com.fasterxml.jackson.jaxrs:jackson-jaxrs-json-provider:2.12.2',
+            'com.fasterxml.jackson.module:jackson-module-jaxb-annotations:2.12.2'
```
Comment (twincitiesguy) [inline build.gradle:64]: I don't see the actual jersey-client library here (`com.sun.jersey:jersey-client:1.19.4`), and we definitely use that - it gets pulled in as a transient, but we should declare it

**Review by twincitiesguy — CHANGES_REQUESTED** [review]

File: build.gradle (hunk 7 above)
Comment (shy950521) [reply build.gradle]: done

**Review by twincitiesguy — APPROVED** [review]

**Pull request merged**